authentication/models/permissions.py

```diff
@@ -0,0 +1,59 @@
+"""
+Models (unmanaged) for custom app permissions.  These must remain within the
+same app for the permissions architecture to function properly.
+"""
```

Comment:
Just a note - I personally encountered problems in testing when I had models for a given app split across multiple files (unless I imported all from one file into the other).  I was only able to get this to work when all of the app's models were in a single file, or similarly, if I imported all from one into the other.